### PR TITLE
Igmguesses IO in IGMSightline

### PR DESCRIPTION
**Pull request by ntejos**
- Updates to IGMSightlines for IGMGuesses IO capabilities.
- Includes updates to IGMGuesses itself

### CODE REVIEW

**Comment by profxj** [comment]
First, let's clear #175 as it helps with tests and
eliminates all of that astropy-helpers junk.

That said, I am reviewing this PR now.

File: pyigm/igm/igmsightline.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def from_json(cls, jsonfile, **kwargs):
         return slf
 
     @classmethod
-    def from_igmguesses(cls, radec, zem, igmgfile, name=None, **kwargs):
+    def from_igmguesses_old(cls, radec, zem, igmgfile, name=None, **kwargs):
```
Comment (profxj) [inline pyigm/igm/igmsightline.py]: I think you can remove this method and have this
functionality in the new one by making radec and zem
optional keywords, e.g.

radec=None, zem=None

and then

``` 
if radec is not None:
         blah blah
```

File: pyigm/igm/igmsightline.py [hunk 2]
```diff
@@ -133,10 +173,90 @@ def make_igmsystems(self, igmsystem=None, **kwargs):
...
+            out_dict['cmps'][key]['reliability'] = str(comp.reliability)
+            out_dict['cmps'][key]['comment'] = str(comp.comment)
+            # out_dict['cmps'][key]['mask_abslines'] = comp.mask_abslines
+
+        # JSONify
+        gd_dict = ltu.jsonify(out_dict)
+
+        # Write file
+        # with io.open(outfile, 'w', encoding='utf-8') as f:
+        f = open(outfile, 'w')
```
Comment (profxj) [inline pyigm/igm/igmsightline.py]: use  ltu.savejson() instead

**Review by profxj — APPROVED** [review]
just a few minor edits requested.

**Comment by ntejos** [comment]
Ok, waiting now for PR #175 before attempting merging.

**Comment by profxj** [comment]
PR #175 is cleared.  You might pull master into this
branch and make sure the tests all pass before merging.

**Comment by ntejos** [comment]
Roger. Will see what the tests say...

**Comment by ntejos** [comment]
Test failing (partially) because it depends on linetools' PR #458.

**Comment by profxj** [comment]
Have cleared the linetools PR.  Check again
when you can.

**Comment by ntejos** [comment]
Failing in numpy 1.14 when comparing to the written file with the reference file. In my computer there is no such an error, neither in the other versions of numpy. May be the sorting of the components once loaded and then written. Will check by modifying linetools to give a more meaningful message.

**Comment by ntejos** [comment]
Failing some tests that are unrelated to this PR. 

Also failing 1 test in numpy 1.14 where the written file has a different number for 'wrest' than the reference. This is most likely coming from the fact that `AbsLines` inside the list `AbsComponent._abslines` are sorted in a different way depending on the numpy version. We could fix this by imposing that those abslines are sorted in a very specific way in linetools, but I think is beyond the scope of this PR. Would like to merge if you agree.

**Comment by ntejos** [comment]
@profxj may we clear this as it is?

**Comment by profxj** [comment]
Merging.

**Pull request merged**